pkg/provider/aws/provider.go

```diff
@@ -60,8 +61,11 @@ func NewProvider() (provider.Provider, error) {
 	}, nil
 }
 
-func retrieveInstanceNetworkInterfaceMacAddress() (string, error) {
-	res, err := http.Get(instanceMetadataEndpoint + "/mac")
+func retrieveInstanceNetworkInterfaceMacAddress(token string) (string, error) {
+	client := &http.Client{}
+	req, _ := http.NewRequest(http.MethodGet, instanceMetadataEndpoint+"/mac", nil)
```

Comment:
Why not checking errors ?

Reply:
Because this is a wrapper function that is not actually making any requests, the request is client.Do(req)

Reply:
Yes, but i could send errors